cloud/services/groups/groups.go

```diff
@@ -56,18 +56,18 @@ func (s *Service) Reconcile(ctx context.Context) error {
 
 // Delete deletes the resource group with the provided name.
 func (s *Service) Delete(ctx context.Context) error {
-	managed, err := s.isGroupManaged(ctx)
+	managed, err := s.IsGroupManaged(ctx)
 	if err != nil {
 		return errors.Wrap(err, "could not get resource group management state")
 	}
 
 	if !managed {
-		s.Scope.V(4).Info("Skipping resource group deletion in unmanaged mode")
-		return nil
+		s.Scope.V(2).Info("Should not delete resource group in unmanaged mode")
+		return azure.ErrNotOwned
 	}
 
 	s.Scope.V(2).Info("deleting resource group", "resource group", s.Scope.ResourceGroup())
-	err = s.Client.Delete(ctx, s.Scope.ResourceGroup())
+	err = s.Client.DeleteNoWait(ctx, s.Scope.ResourceGroup())
```

Comment:
I can switch this back to Delete() if we prefer to make the RG delete blocking for now

Reply:
I measured ~15 minutes to delete the cluster before this change vs. ~7 minutes to delete the entire resource group with this method.

Reply:
I'm leaning more towards making the RG delete blocking for now, that way we can test this and know if there are issues deleting RGs directly... otherwise, we risk leaking resources if there is a bug and we won't know. Thoughts?